[script.module.codequick] 0.9.5

willforde opened the pull request:
### Description
* Mostly documentation changes.
* Added Time to live option to persistent storage.
* Added support for fetching localized strings using "msgid" as the reference. i.e. gettext
* Added some extra functions to utils, and removed some others.
* Lots of other minor changes.
* Tested with Kodi v18 Leia.

Sorry in advanced for the large commit changes log.

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

enen92 commented on script.module.codequick/lib/codequick/storage.py:
```diff
@@ -89,70 +94,119 @@ def flush(self):
             self._stream.write(content)
             self._hash = current_hash
             self._stream.flush()
+        else:
```
not needed

enen92 commented on script.module.codequick/lib/codequick/support.py:
```diff
@@ -86,52 +83,93 @@ def emit(self, record):
+            addon_path = xbmc.translatePath(addon_path)
+
+        # Location of strings.po file
+        strings_po = os.path.join(addon_path, "resources", "language", "resource.language.en_gb", "strings.po")
```
you're not allowed to directly parse/mess with those files. Use the API only please. I know you're trying to add support for strings instead of numeric id's but nevertheless, you need to use another approach that does not rely on reading the strings.po file directly

willforde commented:
I have removed the parsing code, and changed it so that the user can manually add string references using `utils.string_map`.
I also took this chance to re-order methods in storage.py to remove duplicate code.

enen92 commented:
Thanks much for removing the strings stuff! Merging
